src/debugging/renderdoc.cpp

```diff
@@ -17,22 +18,21 @@
 #endif
 
 namespace nova::renderer {
-    RENDERDOC_API_1_3_0* load_renderdoc(const std::string& renderdoc_dll_path) {
+    result<RENDERDOC_API_1_3_0*> load_renderdoc(const std::string& renderdoc_dll_path) {
 #if defined(NOVA_WINDOWS)
-        HINSTANCE const renderdoc_dll = LoadLibrary(renderdoc_dll_path.c_str());
+        using HINSTANCE = HINSTANCE__* const;
+        HINSTANCE renderdoc_dll = LoadLibrary(renderdoc_dll_path.c_str());
         if(!renderdoc_dll) {
             const std::string error = get_last_windows_error();
-            NOVA_LOG(ERROR) << "Could not load RenderDoc. Error: " << error;
-
-            return nullptr;
+            return result<RENDERDOC_API_1_3_0*>(MAKE_ERROR("Could not load RenderDoc. Error: {:s}", error));
```

Comment:
You could probably drop the template argument if you added some deduction guides so that CTAD worked.